bukkit/rpk-chat-bukkit/src/main/kotlin/com/rpkit/chat/bukkit/command/setchatnamecolor/SetChatNameColorCommand.kt

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+package com.rpkit.chat.bukkit.command.setchatnamecolor
+
+import com.rpkit.chat.bukkit.RPKChatBukkit
+import com.rpkit.chat.bukkit.database.table.RPKChatNameColorTable
+import com.rpkit.core.service.Services
+import com.rpkit.players.bukkit.profile.minecraft.RPKMinecraftProfileId
+import com.rpkit.players.bukkit.profile.minecraft.RPKMinecraftProfileService
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+import org.bukkit.entity.Player
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletableFuture.runAsync
+
+/**
+ * SetChatNameColor command.
+ * Sets the chat color name for a player.
+ */
+class SetChatNameColorCommand(private val plugin: RPKChatBukkit) : CommandExecutor {
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (!sender.hasPermission("rpkit.chat.command.setchatnamecolor")) {
+            sender.sendMessage(plugin.messages["no-permission-setchatnamecolor"])
+            return true
+        }
+        if (sender !is Player) {
+            sender.sendMessage(plugin.messages["not-from-console"])
+            return true
+        }
+        val minecraftProfileService = Services[RPKMinecraftProfileService::class.java]
+        if (minecraftProfileService == null) {
+            sender.sendMessage(plugin.messages["no-minecraft-profile-service"])
+            return true
+        }
+        val minecraftProfile = minecraftProfileService.getPreloadedMinecraftProfile(sender)
+        if (minecraftProfile == null) {
+            sender.sendMessage(plugin.messages["no-minecraft-profile"])
+            return true
+        }
+
+        // if no arguments
+        if (args.isEmpty()) {
+            sender.sendMessage(plugin.messages["setchatnamecolor-usage"])
+            return true
+        }
+
+        // retrieve desired chat name color
+        val chatNameColor = args[0]
+        if (isInputTooLong(chatNameColor)) {
+            sender.sendMessage(plugin.messages["setchatnamecolor-too-long"])
+            return true
+        }
+
+        if (!isHexColorCodeValid(chatNameColor)) {
+            sender.sendMessage(plugin.messages["setchatnamecolor-invalid-color-code"])
+            return true
+        }
+
+        val minecraftProfileId = minecraftProfile.id
+        if (minecraftProfileId == null) {
+            sender.sendMessage(plugin.messages["no-minecraft-profile"])
+            return true
+        }
+
+        setChatNameColorAsync(minecraftProfileId, chatNameColor).thenRun {
+            sender.sendMessage(plugin.messages["setchatnamecolor-chat-name-color-has-been-set"])
+        }.exceptionally { exception ->
+            plugin.logger.severe("Failed to set chat name color for ${minecraftProfile.name}")
+            plugin.logger.severe(exception.message)
```

Comment:
```suggestion
            plugin.logger.log(SEVERE, "Failed to set chat name color for ${minecraftProfile.name}", exception)
```
(May need SEVERE to be imported but we can do this in one log)

Reply:
Made this change!